File: eid-mobile-android/data/src/main/java/com/digitall/eid/data/models/network/certificates/response/CertificateDetailsResponse.kt
/**
 * Please follow code style when editing project
 * Please follow principles of clean architecture
 * Created 2024 by Roman Kryvolapov
 **/
package com.digitall.eid.data.models.network.certificates.response

import com.google.gson.annotations.SerializedName

data class CertificateDetailsResponse(
    @SerializedName("id") val id: String?,
    @SerializedName("status") val status: String?,
    @SerializedName("eidAdministratorId") val eidAdministratorId: String?,
    @SerializedName("eidAdministratorOfficeId") val eidAdministratorOfficeId: String?,
    @SerializedName("eidAdministratorName") val eidAdministratorName: String?,
    @SerializedName("eidentityId") val eidentityId: String?,
    @SerializedName("commonName") val commonName: String?,
    @SerializedName("validityFrom") val validityFrom: String?,
    @SerializedName("validityUntil") val validityUntil: String?,
    @SerializedName("createDate") val createDate: String?,
    @SerializedName("serialNumber") val serialNumber: String?,
    @SerializedName("deviceId") val deviceId: String?,
    @SerializedName("levelOfAssurance") val levelOfAssurance: String?,
    @SerializedName("reasonId") val reasonId: String?,
    @SerializedName("reasonText") val reasonText: String?,
    @SerializedName("isExpiring") val expiring: Boolean?,
    @SerializedName("alias") val alias: String?
)

File: eid-mobile-ios/eID-Mobile-iOS/02_Data/Entities/ElectronicIdentity/Certificates/CertificateDetailsResponse.swift
//
//  CertificateDetailsResponse.swift
//  eID-Mobile-iOS
//
//  Created by Sunay Gyultekin on 16.02.24.
//

import Foundation


struct CertificateDetailsResponse: Codable {
    // MARK: - Properties
    var id: String?
    var applicationNumber: String?
    var status: CertificateStatus?
    var eidAdministratorId: String?
    var eidAdministratorOfficeId: String?
    var eidAdministratorName: String?
    var eidentityId: String?
    var commonName: String?
    var validityFrom: String?
    var validityUntil: String?
    var createDate: String?
    var serialNumber: String?
    var deviceId: String?
    var levelOfAssurance: CertificateLevelOfAssurance?
    var reasonId: String?
    var reasonText: String?
    var isExpiring: Bool?
    var alias: String?
}
